### PR TITLE
Main Menu

### PR DESCRIPTION
**Pull request by richiestuver**
# TLDR;
- Start game with Main Menu and Persist scenes
- Run game - game will automatically start from main menu scene
- All menu navigation is implemented
- Can access editor for both new and existing boards
- Play Game flow loads a board but dumps you into the sample Play Game scene for now

**Note**: The VS Code formatter for C# auto-formats function calls to start the opening bracket on a new line. You'll see some files with only changes being a bracket moved to new line only. 

# Summary of Changes
**Menu Flow Logic is complete**. Properly loads save data for boards, displays to user, enter the editor with the appropriate board data, enter the Play Game scene with proper board ID. Standardized naming in scene for Menus and Buttons. Added additional prefabs for Back button, etc.

- Main Menu Scene
    - Play
        - Player Select Menu 
        - Board Select Menu
            - Populated with existing save data
            - Select a board from the list (scrollable)
            - Launch the "Play Game" Scene (currently a sample)
    - Editor
        - New Board > Editor
        - Load Board > Board Select Menu > Editor
    - Quit

# To Do
- Piece selection logic on Player Select Menu
- Generate the board in Play Game scene based on save data
- Pause menu
- Update menu visuals based on designs

# Files Modified
- Save Controller
    - Moved SetBoardID to be a public method of SaveController for use across UI scripts
    - Auto-formatting of function call brackets 🤷🏼 
- Scene Controller
    - Auto-formatted function call brackets 🤷🏼 no actual code changes
- Main Menu **(most changes are here)**
    - Using method calls from SceneController (e.g GoToEditor) instead of working with SceneManager directly
    - Use Stack data structure for efficiently pushing and popping menus
    - Method wrappers for navigating menus easily (PushMenu, PopMenu)
    - Added callback functions for all menus
    - Standardize naming scheme for Menus and Callbacks
- Board List
    - Retrieve save data from save controller 
    - Dynamically populate board list with board names 
    - Calculate index in list and set board ID accordingly

### CODE REVIEW

**Review by jonathanLum — APPROVED** [review]
looks awesome
